src/handlers/inbound/services/{ServiceType}.ts

```diff
@@ -0,0 +1,58 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License")
+ and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed
+ on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Kevin Leyow <kevin.leyow@modusbox.com>
+
+ --------------
+ ******/
+import { Request, ResponseObject } from '@hapi/hapi'
+import { StateResponseToolkit } from '~/server/plugins/state'
+import { PISPPrelinkingModel } from '~/models/outbound/pispPrelinking.model';
+import { Message } from '~/shared/pub-sub'
+import { Enum } from '@mojaloop/central-services-shared';
+import { ServiceType } from '~/models/outbound/pispPrelinking.interface';
+
+
+/**
+ * Handles an inbound `PUT /services/{ServiceType}` request
+ */
+async function put (_context: unknown, request: Request, h: StateResponseToolkit): Promise<ResponseObject> {
+  const serviceType = request.params.ServiceType
+
+  if (serviceType == ServiceType.THIRD_PARTY_DFSP) {
+    const channel = PISPPrelinkingModel.notificationChannel(serviceType)
+    const pubSub = h.getPubSub()
+
+    // don't await on promise to resolve, let finish publish in background
+    setImmediate(async () => {
+      pubSub.publish(channel, request.payload as unknown as Message)
```

Comment:
instead of using pubSub.publish - I would use `deferredJob.trigger` method instead -> then all handlers will only depend on your model -> the purest way would be to add a static method to your model class which wraps around deferredJob.trigger invocation -> then your pub/sub code will be used only in your model

Reply:
done